test/e2e/lb_subnets_test.go

```diff
@@ -0,0 +1,450 @@
+
+	// Verify we can reach the CLB with the provided public subnets.
+	externalTestPodName := types.NamespacedName{Name: icName.Name + "-external-verify", Namespace: icName.Namespace}
+	testHostname := "apps." + ic.Spec.Domain
+	t.Logf("verifying external connectivity for ingresscontroller %q using an CLB with specified public subnets", ic.Name)
+	verifyExternalIngressController(t, externalTestPodName, testHostname, testHostname)
+
+	// Now, update the IngressController to use invalid (non-existent) subnets.
+	t.Logf("updating ingresscontroller %q to use invalid subnets", ic.Name)
+	invalidSubnets := &operatorv1.AWSSubnets{
+		Names: []operatorv1.AWSSubnetName{"subnetfoo", "subnetbar"},
+	}
+	if err = updateIngressControllerWithRetryOnConflict(t, icName, 5*time.Minute, func(ic *operatorv1.IngressController) {
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters = &operatorv1.ProviderLoadBalancerParameters{
+			Type: operatorv1.AWSLoadBalancerProvider,
+			AWS: &operatorv1.AWSLoadBalancerParameters{
+				Type: operatorv1.AWSClassicLoadBalancer,
+				ClassicLoadBalancerParameters: &operatorv1.AWSClassicLoadBalancerParameters{
+					Subnets: invalidSubnets,
+				},
+			},
+		}
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// Wait for the LoadBalancerProgressing to become True, which represents the message telling
+	// the cluster admin to delete the service to effectuate the subnets.
+	loadBalancerProgressingTrue := operatorv1.OperatorCondition{
+		Type:   ingress.IngressControllerLoadBalancerProgressingConditionType,
+		Status: operatorv1.ConditionTrue,
+	}
+	if err = waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, loadBalancerProgressingTrue); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Since the subnets are invalid, the load balancer will fail to provision and set LoadBalancerReady=False.
+	loadBalancerReadyFalse := operatorv1.OperatorCondition{
+		Type:   operatorv1.LoadBalancerReadyIngressConditionType,
+		Status: operatorv1.ConditionFalse,
+	}
+	effectuateIngressControllerSubnets(t, ic, invalidSubnets, loadBalancerReadyFalse)
+
+	// Now, update the IngressController to change to use a NLB and the public subnets again.
+	t.Logf("updating ingresscontroller %q to use an NLB and public subnets", ic.Name)
+	if err = updateIngressControllerWithRetryOnConflict(t, icName, 5*time.Minute, func(ic *operatorv1.IngressController) {
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type = operatorv1.AWSNetworkLoadBalancer
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters = &operatorv1.AWSNetworkLoadBalancerParameters{
+			Subnets: publicSubnets,
+		}
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	if err = waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, loadBalancerProgressingTrue); err != nil {
```

Comment:
do we need this as you are already checking Progressing=True in effectuateIngressControllerSubnets function ?

Reply:
Good catch! No we don't. Removed.